src/charm.py

```diff
@@ -464,6 +464,9 @@ def _on_peer_relation_changed(self, event: HookEvent) -> None:
 
         self.postgresql_client_relation.update_read_only_endpoint()
 
+        if self.unit.is_leader() and "init-pgbackrest" in self.app_peer_data:
```

Comment:
Should we move the condition inside the `check_stanza` function to not spread the `init-pgbackrest` logic between many different functions?